pkg/controller/jobframework/reconciler.go

```diff
@@ -164,6 +182,32 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	// 1.1 If the workload is pending deletion, suspend the job if needed
+	// and drop the finalizer.
+	if wl != nil && !wl.DeletionTimestamp.IsZero() {
+		log.V(2).Info("The workload is marked for deletion")
+		if !job.IsSuspended() {
+			err := r.stopJob(ctx, job, object, wl, "Workload is deleted")
+			if err != nil {
+				log.Error(err, "Suspending job with deleted workload")
+			}
+			return ctrl.Result{}, err
+		} else {
+			// make sure the finalizer is removed
+			err := r.setFinalizer(ctx, wl, false)
```

Comment:
I would like to have a dedicated function, such as  `removeFinalizer`. Also, would we need to remove the finalizer here, if we added the finalizer removal to the workload_controller, that deletes the finalizer, if the job is suspended?